plugins/xterm/README.md

```diff
@@ -1,3 +1,48 @@
-# Xterm plugin
+### Plugin xterm
 
-This plugin sets the xterm title for the current directory and the currently running command.
+#### Description
+
+This script automatically sets your xterm title with host and location information. It dynamically updates the title to reflect the current user, host, directory, and command being executed.
+
+#### Source
+
+You can find the original source of this script [here](https://github.com/Bash-it/bash-it/blob/bf2034d13d/plugins/available/xterm.plugin.bash).
+
+#### Variables
+
+- `PROMPT_CHAR` (optional): This variable is shared with powerline.
+- `OMB_PLUGIN_XTERM_SHORT_TERM_LINE` (optional): Controls whether to shorten the directory name in the title.
+- `OMB_PLUGIN_XTERM_SHORT_USER` (optional): Overrides the default user name.
+- `OMB_PLUGIN_XTERM_SHORT_HOSTNAME` (optional): Overrides the default hostname.
+
+#### Functions
+
+1. **`_omb_plugin_xterm_set_title`**
+
+   - Sets the xterm title with the provided string.
+
+2. **`_omb_plugin_xterm_precmd_title`**
+
+   - Updates the xterm title before executing a command.
+   - Includes the user, host, directory, and prompt character.
+
+3. **`_omb_plugin_xterm_preexec_title`**
+
+   - Updates the xterm title after executing a command.
+   - Includes the user, host, directory, and the first part of the executed command.
```

Comment:
I believe these are internal functions and not a part of the public interface.

Reply:
You're cool, I thought they could be used externally

Reply:
Technically, it is possible to use those functions externally, but it's not intended. It is important to draw a border between the private and public interface (even though we cannot programmatically prevent the external shell functions from using our private functions). In that way, we can maintain compatibility with the future changes. If everything is considered public, we cannot change anything in the future. A certain part needs to be left as a private interface to keep room for future changes.